src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -306,6 +306,8 @@ export function ComboBoxBase<O, V extends Value>(props: ComboBoxBaseProps<O, V>)
 
   // Reset inputValue when closed or selected changes
   useEffect(() => {
+    // if we are using the onSearch functionality with debounceSearch, we don't want to reset the input value
+    if (onSearch && debouncedSearch) return;
```

Comment:
question: Do we need to check against `debouncedSearch` too? `onSearch` should be enough?

Reply:
yes, good question! I originally had it without `debounced`, but when I was testing, I noticed after I made a selection and hit the "clear button" to clear the filter, it wouldn't reset so I added the extra check